lib/core/src/recover/handlers/tests/handle_send_swap_tests_integration.rs

```diff
@@ -0,0 +1,454 @@
+    ) -> RecoveryContext {
+        let tx_id = Txid::from_str(tx_id_hex).unwrap();
+
+        // Create history tx
+        let history_tx = HistoryTxId {
+            txid: tx_id,
+            height: height as i32,
+        };
+
+        // Add to script history map
+        let mut script_history = context
+            .lbtc_script_to_history_map
+            .get(script)
+            .cloned()
+            .unwrap_or_default();
+        script_history.push(history_tx.clone());
+        context
+            .lbtc_script_to_history_map
+            .insert(script.clone(), script_history);
+
+        // Create wallet tx
+        let wallet_tx = create_mock_lbtc_wallet_tx(tx_id_hex, height, -100000); // Negative amount for outgoing
+
+        // Add to outgoing tx map
+        context.tx_map.outgoing_tx_map.insert(tx_id, wallet_tx);
+
+        context
+    }
+
+    // Helper to add a claim tx to the history without adding it to the wallet tx map
+    fn add_claim_tx_to_context(
+        mut context: RecoveryContext,
+        script: &Script,
+        tx_id_hex: &str,
+        height: u32,
+    ) -> RecoveryContext {
+        let tx_id = Txid::from_str(tx_id_hex).unwrap();
+
+        // Create history tx
+        let history_tx = HistoryTxId {
+            txid: tx_id,
+            height: height as i32,
+        };
+
+        // Add to script history map
+        let mut script_history = context
+            .lbtc_script_to_history_map
+            .get(script)
+            .cloned()
+            .unwrap_or_default();
+        script_history.push(history_tx.clone());
+        context
+            .lbtc_script_to_history_map
+            .insert(script.clone(), script_history);
+
+        context
+    }
+
+    // Helper to add an incoming transaction (refund) to the recovery context
+    fn add_incoming_tx_to_context(
+        mut context: RecoveryContext,
+        script: &Script,
+        tx_id_hex: &str,
+        height: u32,
+        amount: u64,
+    ) -> (RecoveryContext, WalletTx) {
+        let tx_id = Txid::from_str(tx_id_hex).unwrap();
+
+        // Create history tx
+        let history_tx = HistoryTxId {
+            txid: tx_id,
+            height: height as i32,
+        };
+
+        // Add to script history map
+        let mut script_history = context
+            .lbtc_script_to_history_map
+            .get(script)
+            .cloned()
+            .unwrap_or_default();
+        script_history.push(history_tx.clone());
+        context
+            .lbtc_script_to_history_map
+            .insert(script.clone(), script_history);
+
+        // Create wallet tx
+        let wallet_tx = create_mock_lbtc_wallet_tx(tx_id_hex, height, amount as i64); // Positive amount for incoming
+
+        // Add to incoming tx map
+        context
+            .tx_map
+            .incoming_tx_map
+            .insert(tx_id, wallet_tx.clone());
+
+        (context, wallet_tx)
+    }
+
+    fn create_empty_transaction() -> Transaction {
```

Comment:
Could reuse `tests::test::create_empty_lbtc_transaction`

Reply:
Done

Reply:
Did you push?

Reply:
Just now. Thanks.